cache/cache.go

```diff
@@ -197,8 +237,9 @@ func serveWatchEvents(ctx context.Context, watchCtx *watchCtx) {
 			clientv3.WithCreatedNotify(),
 		}
 		if oldestRev := watchCtx.cache.demux.PeekOldest(); oldestRev != 0 {
-			opts = append(opts,
-				clientv3.WithRev(oldestRev+1))
+			opts = append(opts, clientv3.WithRev(oldestRev+1))
+		} else if watchCtx.startRev != 0 {
+			opts = append(opts, clientv3.WithRev(watchCtx.startRev))
```

Comment:
This is wrong you first should Get, and then Watch, so the demux should not matter here anymore.